tools/junit-openfeature/README.md

```diff
@@ -0,0 +1,156 @@
+# JUnit Open Feature extension
+
+A JUnit5 extension to reduce boilerplate code for testing code which utilizes OpenFeature.
+
+## Getting Started
+
+We are supporting two different flavors for testing, a [simple](#simple-configuration) and an [extended](#extended-configuration) configuration.
+      
+Notice: We are most likely not multithread compatible!
+### Simple Configuration
+
+Choose the simple configuration if you are only testing in one domain.
+Per default, it will be used in the global domain.
+
+```java
+@Test
+@Flag(name = "BOOLEAN_FLAG", value = "true")
+void test() {
+    // your test code
+}
+``` 
+ 
+#### Multiple flags
+
+The `@Flag` annotation can be also repeated multiple times.
+
+```java
+@Test
+@Flag(name = "BOOLEAN_FLAG", value = "true")
+@Flag(name = "BOOLEAN_FLAG2", value = "true")
+void test() {
+    // your test code
+}
+``` 
+
+#### Defining Flags for a whole test-class
+
+`@Flags` can be defined on the class-level too, but method-level
+annotations will superseded class-level annotations.
+
+```java
+@Flag(name = "BOOLEAN_FLAG", value = "true")
+@Flag(name = "BOOLEAN_FLAG2", value = "false")
+class Test {
+    @Test
+    @Flag(name = "BOOLEAN_FLAG2", value = "true") // will be used
+    void test() {
+        // your test code
+    }
+}
+``` 
+
+#### Setting a different domain
+
+You can define an own domain on the test-class-level with `@OpenFeatureDefaultDomain` like:
+
+```java
+@OpenFeatureDefaultDomain("domain")
+class Test {
+    @Test
+    @Flag(name = "BOOLEAN_FLAG", value = "true")
+        // this flag will be available in the `domain` domain
+    void test() {
+        // your test code
+    }
+}
+```
+
+### Extended Configuration
+
+Use the extended configuration when your code needs to use multiple domains.
+
+```java
+@Test
+@OpenFeature({
+    @Flag(name = "BOOLEAN_FLAG", value = "true")
+})
+void test() {
+    // your test code
+}
+```
```

Comment:
I think this example might benefit from seeing the multiple domains you mention.

Reply:
TBH you can maybe forgo this example entirely since the very last one covers it well.

Reply:
agreed, but i think we should start the docs with a simple example - hence that i might move the last one to the top